GenericsHomework/Node.cs

```diff
@@ -0,0 +1,120 @@
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GenericsHomework;
+public class Node<T>
+{
+    public T Value { get; }
+    public Node<T> Next { get; private set; }
+
+    public Node(T value)
+    {
+        Value = value;
+        Next = this;
+    }
+
+    public void SetNext(Node<T> next)
+    {
+        // I think having the check has made it so that next cannot be null
+        ArgumentNullException.ThrowIfNull(next);
+        this.Next = next;
+    }
+
+    public override string ToString()
+    {
+        Node<T> dummy = new(this.Value)
+        {
+            Next = this
+        };
+        string ret = "";
+        // goes through linked list starting at node called
+        while (dummy.Next.Next != this)
+        {
+            if (dummy.Next.Value != null)
+            {
+
+                ret += dummy.Next.Value.ToString() + " ";
+            }
+            dummy.Next = dummy.Next.Next;
+
+        }
+        if (dummy.Next.Value != null)
+        {
+
+            ret += dummy.Next.Value.ToString();
+        }
+
+        return ret;
+    }
+    public void Append(T value)
+    {
+        ArgumentNullException.ThrowIfNull(value);
+        if (Exists(value))
+        {
+            return;
+        }
+        Node<T> dummy = new(value)
+        {
+            Next = this
+        };
+        while (dummy.Next.Next != this)
+        {
+            dummy.Next = dummy.Next.Next;
+        }
+
+        Node<T> newNode = new(value);
+        dummy.Next.Next = newNode;
+        newNode.Next = this;
+    }
+    public void Clear()
+    {
+
+        Node<T> current = this;
+        Node<T> temp = current.Next;
+
+        while (temp != current)
+        {
+            Node<T> nextTemp = temp.Next;
+            temp.Next = this; // Disconnect the node
+            temp = nextTemp;
+        }
+
+        current.Next = current;
+        //All we need to do to clear the list is reconnect the next to the node itself. 
+        //Since the other nodes are disconnected from the head node after this method executes, the remaining nodes are eligible for garbage collection
+        //Since there is nothing to reference them, we do not need to worry about what happens to them. They are taken care of in garbage collection
+    }
+    public bool Exists(T value)
+    {
+        ArgumentNullException.ThrowIfNull(value);
+        Node<T> dummy = new(value)
+        {
+            Next = this
+        };
+        while (dummy.Next.Next != this)
+        {
+            if (dummy.Next.Value != null)
+            {
+                if (dummy.Next.Value.Equals(value))
+                {
+                    return true;
+                }
+            }
+            dummy.Next = dummy.Next.Next;
+
+        }
+        if (dummy.Next.Value != null)
+        {
+            if (dummy.Next.Value.Equals(value))
+            {
+                return true;
+            }
+        }
+        return false;
```

Comment:
```suggestion
        var node = this;
        do
        {
            if (EqualityComparer<T>.Default.Equals(node.Value, value)) return true;
            node = node.Next;
        } while (node != this);
        return false;
```
This seems over complicated and error prone. And what about if value or Value is null? Why not something like this?